internal/subnamespace/sync.go

```diff
@@ -404,12 +405,16 @@ func IsUpdateNeeded(sns client.Object, childrenRequests []danav1.Namespaces, all
 }
 
 // NamespacesEqual gets two []danav1.Namespaces and returns whether they are equal.
-func NamespacesEqual(nsA, nsB []danav1.Namespaces) bool {
+func NamespacesEqual(nsA, nsB []danav1.Namespaces, snsContext context.Context, snsClient client.Client) bool {
```

Comment:
Also here call it `ctx` and `k8sClient` and move it to the front of the signature